python/ray/experimental/client/api.py

```diff
@@ -131,10 +166,136 @@ def call_remote(self, instance: "ClientStub", *args, **kwargs):
+
+        Returns:
+            Information about the Ray clients in the cluster.
+        """
+        return self.worker.get_cluster_info(
+            ray_client_pb2.ClusterInfoType.NODES)
+
+    def actors(self, actor_id: Optional[str] = None):
+        """Fetch actor info for one or more actor IDs (for debugging only).
+
+        Args:
+            actor_id: A hex string of the actor ID to fetch information about. If
+                this is None, then all actor information is fetched.
+
+        Returns:
+            Information about the actors.
+        """
+        client_id = None
+        if actor_id is not None:
+            client_id = bytes.fromhex(actor_id)
+
+        return self.worker.get_cluster_info(
+            ray_client_pb2.ClusterInfoType.ACTORS, client_id=client_id)
+
+    def objects(self, object_ref: Optional[str] = None):
+        """Fetch and parse the object table info for one or more object refs.
+
+        Args:
+            object_ref: An object ref to fetch information about. If this is None,
+                then the entire object table is fetched.
+
+        Returns:
+            Information from the object table.
+        """
+        client_id = None
+        if object_ref is not None:
+            client_id = bytes.fromhex(object_ref)
+
+        return self.worker.get_cluster_info(
+            ray_client_pb2.ClusterInfoType.OBJECTS, client_id=client_id)
+
+    def cluster_resources(self):
+        """Get the current total cluster resources.
+
+        Note that this information can grow stale as nodes are added to or removed
+        from the cluster.
+
+        Returns:
+            A dictionary mapping resource name to the total quantity of that
+                resource in the cluster.
+        """
+        return self.worker.get_cluster_info(
+            ray_client_pb2.ClusterInfoType.CLUSTER_RESOURCES)
+
+    def available_resources(self):
+        """Get the current available cluster resources.
+
+        This is different from `cluster_resources` in that this will return idle
+        (available) resources rather than total resources.
+
+        Note that this information can grow stale as tasks start and finish.
+
+        Returns:
+            A dictionary mapping resource name to the total quantity of that
+                resource in the cluster.
+        """
+        return self.worker.get_cluster_info(
+            ray_client_pb2.ClusterInfoType.AVAILABLE_RESOURCES)
+
+    @property
+    def state(self) -> "ClientStateAPI":
+        return ClientStateAPI(self)
+
     def __getattr__(self, key: str):
         if not key.startswith("_"):
             raise NotImplementedError(
                 "Not available in Ray client: `ray.{}`. This method is only "
                 "available within Ray remote functions and is not yet "
                 "implemented in the client API.".format(key))
         return self.__getattribute__(key)
+
+
+class ClientStateAPI:
```

Comment:
Let's not expose ray.state for now--- it's an undocumented API

Reply:
That'd be fine, except `ray.nodes()` -- as was requested in the issue -- is part of the state API. Ditto also `cluster_resources()` and `available_resources()`. Let's sync on this.